packages/common/src/logger-base.ts

```diff
@@ -0,0 +1,186 @@
+    if (this.logLevel > LogLevel.debug) {
+      return;
+    }
+    this.log(LogStyle.info, content);
+  }
+
+  public imp(content: string | Record<string, unknown> | Error): void {
+    if (this.logLevel > LogLevel.important) {
+      return;
+    }
+    this.log(LogStyle.imp, content);
+  }
+
+  public warn(content: string | Record<string, unknown> | Error): void {
+    this.log(LogStyle.warn, content);
+  }
+
+  public err(content: string | Record<string, unknown> | Error): void {
+    this.log(LogStyle.err, content);
+  }
+
+  public log(style: LogStyle, content: string | Record<string, unknown> | Error): void {
+    throw new Error('Not implmented in base class')
+  }
+}
+
+export interface ServiceLog {
+  [key: string]: any;
+}
+
+/**
+ * Logs content from app services to files and/or console
+ */
+export class ServiceLoggerBase {
+  public readonly service: string;
+  public readonly logToFiles: boolean
+  public readonly logToConsole: boolean
+  public readonly logLevel: LogLevel
+
+  protected readonly stream: WriteableStream;
+
+
+  constructor(service: string, logLevel: LogLevel, logToFiles: boolean, stream?: WriteableStream) {
+    this.service = service;
+    this.logLevel = logLevel
+    this.logToFiles = logToFiles
+    this.stream = stream
+
+    this.logToConsole = this.logLevel <= LogLevel.debug
+  }
+
+  /**
+   * Converts the service log to logfmt and writes it to `this.filePath`
+   * @param serviceLog Service log object
+   */
+  public log(serviceLog: ServiceLog): void {
+    this.write(this.format(serviceLog))
+  }
+
+  /**
+   * Writes the log message to the file stream and/or the console based on the config.
+   * @param message Content to log
+   */
+  public write(message: string): void {
+    const now = new Date();
+    // RFC1123 timestamp
+    message = `[${now.toUTCString()}] service=${this.service} ${message}`;
+
+    if (this.logToConsole) {
+      console.log(message);
+    }
+  }
+
+  /**
+   * Converts `serviceLog` key/value object to logfmt
+   * @param serviceLog Service log object
+   */
+  public format(serviceLog: ServiceLog): string {
+    throw new Error('Not implmented in base class')
+  }
+}
+
+export interface FileLoggerFactory {
+    (logPath: string): WriteableStream;
+}
+
+export interface LoggerConfig {
+    logDirectory?: string,
+    logLevel?: LogLevel,
+    logToFiles?: boolean,
+}
+
+// Can't have a default log directory as that would require using the `path` module before we know
+// for sure if it's safe to do so. If `logToFiles` is false we must avoid using any node-specific
+// functionality as we might be running in-browser
+const DEFAULT_LOG_CONFIG = {
+    logLevel: LogLevel.important,
+    logToFiles: false
+}
+
+/**
+//  * Global Logger factory
+//  */
+export class LoggerProviderBase {
+  public readonly config: LoggerConfig
+
+  protected readonly _diagnosticLogger: DiagnosticsLoggerBase
+  protected readonly _fileLoggerFactory: FileLoggerFactory
```

Comment:
Ditto, seems like this doesn't belong in the base class if it's not used here anywhere